### PR TITLE
explicitly compare to main when doing change detection for security agent

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

The current base branch is derived automatically by gitlab, and often wrong resulting in the tests being triggered when they shouldn't be. This PR fixes the issue by explicitly asking to compare against main.

See also:
- https://docs.gitlab.com/ee/ci/jobs/job_control.html#jobs-or-pipelines-run-unexpectedly-when-using-changes

This PR also updates the list of watched files to include `pkg/eventmonitor`

Base branch for comparison:
The base branch is currently set to `main`. When `7.yy.x` branch will be created (every 6 weeks), we will do PRs to update the base branch to the correct ones. If this work manually, we will build the automation to update this.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by spikat — APPROVED** [review]

**Review by lebauce — APPROVED** [review]

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -839,12 +839,15 @@ workflow:
...
-      - .gitlab/functional_test/security_agent.yml
+      paths:
+        - pkg/ebpf/**/*
+        - pkg/security/**/*
+        - pkg/eventmonitor/**/*
+        - test/kitchen/site-cookbooks/dd-security-agent-check/**/*
+        - test/kitchen/test/integration/security-agent-test/**/*
+        - test/kitchen/test/integration/security-agent-stress/**/*
+        - .gitlab/functional_test/security_agent.yml
+      compare_to: main
```
Comment (KSerrania) [inline .gitlab-ci.yml:850]: I think that, in the long-term, we'd like this to use a variable (that way, the branch would only need to be set once for every job that uses this). That is not possible yet according to this issue: https://gitlab.com/gitlab-org/gitlab/-/issues/369916, so could we add a TODO here?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -839,12 +839,15 @@ workflow:
...
-      - pkg/ebpf/**/*
-      - pkg/security/**/*
-      - test/kitchen/site-cookbooks/dd-security-agent-check/**/*
-      - test/kitchen/test/integration/security-agent-test/**/*
-      - test/kitchen/test/integration/security-agent-stress/**/*
-      - .gitlab/functional_test/security_agent.yml
+      paths:
+        - pkg/ebpf/**/*
+        - pkg/security/**/*
+        - pkg/eventmonitor/**/*
```
Comment (KSerrania) [inline .gitlab-ci.yml:845]: Is this addition intentional?

**Review by KSerrania — APPROVED** [review]
Left a few minor comments, LGTM.

File: .gitlab-ci.yml (hunk 2 above)
Comment (paulcacheux) [reply .gitlab-ci.yml:845]: yes but I forgot to add it to the PR description, will update it

File: .gitlab-ci.yml (hunk 1 above)
Comment (paulcacheux) [reply .gitlab-ci.yml]: I opened https://gitlab.com/gitlab-org/gitlab/-/merge_requests/126860 to implement variable expansion upstream
